flytekit/common/nodes.py

```diff
@@ -414,3 +414,11 @@ def sync(self):
                 _task_executions.SdkTaskExecution.promote_from_model(te) for te in ne.get_task_executions()
             ]
             # TODO: Sub-workflows too once implemented
+
+    def _sync_closure(self):
+        """
+        Syncs the closure of the underlying execution artifact with the state observed by the platform.
+        :rtype: None
+        """
+        ne = _engine_loader.get_engine().get_node_execution(self)
+        ne.sync()
```

Comment:
so node execution sync() just syncs the closure?  and there's no need to return anything?

Reply:
indeed it does, cool
```
self.sdk_node_execution._closure = client.get_node_execution(self.sdk_node_execution.id).closure
```